compiler/AST/expr.cpp

```diff
@@ -4103,10 +4105,17 @@ GenRet CallExpr::codegen() {
       GenRet   arg        = actual;
 
       if (se && isFnSymbol(se->var)) {
-        arg = codegenCast("chpl_fn_p", arg);
+        if(this->theFnSymbol()->hasFlag(FLAG_EXTERN)) {
+          arg = codegenCast("c_fn_ptr", arg);
+        }
+        else {
+          arg = codegenCast("chpl_fn_p", arg);
+        }
       }
 
+      codegenIsSpecialPrimitive(formal, actual, arg);
```

Comment:
I'd appreciate a comment here that says that codegenIsSpecialPrimitive will replace arg with something else if it takes action.